assignment6/rearranging_cars.py

```diff
@@ -0,0 +1,70 @@
+'''
+Created on 07 July 2017
+
+@author: Eva
+'''
+
+def print_car_moves(initial_car_park, final_car_park):
+    car_moves = rearrange_cars(initial_car_park, final_car_park)
+    
+    for i in range(len(car_moves)):
+        print(car_moves[i].__str__())
+
+def rearrange_cars(initial_car_park, final_car_park):
+    """Rearranges the cars from the initial positions to the end positions.
+    
+    Args:
+        initial_car_park: an array, represents the initial positions of the cars in the parking lot
+        final_car_park: an array, represents the final positions of the cars in the parking lot
+        
+    Returns:
+        an array of Move objects that represent the moves taken to rearrange the cars to their final positions
+    """
+    if len(initial_car_park) != len(final_car_park):
+        raise ValueError("must be the same length: initial_car_park, final_car_park")
+    
+    EMPTY_SPACE = 0
+    # Dictionary used for O(1) lookups when searching for the index of a value in the car_park
+    car_map = {}
+    moves = []
+    
+    for i in range(len(initial_car_park)):
+        car_map[initial_car_park[i]] = i
+    
+    for space in range(len(initial_car_park)):
+        if initial_car_park[space] != final_car_park[space] and initial_car_park[space] != EMPTY_SPACE:
+            empty_space_index = car_map[EMPTY_SPACE]
+            move_car(initial_car_park, car_map, space, empty_space_index, moves)
```

Comment:
I think I would like it better if we will have the list of moves appended in this function instead of doing that in the helper function. We could have the helper function just return a new move and have this statement like:
new_move = move_car(args)
moves.append(new_move)